src/devices/Board/SpiDeviceManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+
+namespace Iot.Device.Board
+{
+    internal class SpiDeviceManager : SpiDevice
+    {
+        private readonly Board _board;
+        private readonly int[] _pins;
+        private SpiDevice _device;
+
+        public SpiDeviceManager(Board board, SpiConnectionSettings connectionSettings, int[]? pins, Func<SpiConnectionSettings, int[], SpiDevice> createOperation)
+        {
+            if (pins == null || pins.Length < 3)
+            {
+                throw new ArgumentException("Must provide a valid set of at least 3 pins", nameof(pins));
+            }
+
+            _board = board;
+            ConnectionSettings = connectionSettings;
+            try
+            {
+                _board.ReservePin(pins[0], PinUsage.Spi, this);
+                _board.ReservePin(pins[1], PinUsage.Spi, this);
+                _board.ReservePin(pins[2], PinUsage.Spi, this);
+                if (pins.Length == 4)
+                {
+                    // The fourth pin, if provided, is the CS line. This may be handled manually, but if done in hardware, SPI
+                    // numbering must be used here I think. So the value would be 0 or 1 for CE0 and CE1 in ALT 0 mode, which are
+                    // the logical pins 7 and 8.
+                    _board.ReservePin(pins[3], PinUsage.Spi, this);
+                }
+
+                _device = createOperation(connectionSettings, pins);
+            }
+            catch (Exception x)
+            {
+                Console.WriteLine($"Exception: {x}");
```

Comment:
You may remove the Console.WriteLine

Reply:
Sure... That's were logging should be used instead.

Reply:
You can create an internal method Log/Info/Error which doesn't do anything and change it once we figure out the convention we want to introduce

Reply:
(or Debug.WriteLine for now)